cst438_gradebook/src/main/java/com/cst438/controllers/GradeBookController.java

```diff
@@ -156,6 +161,46 @@ public void updateGradebook (@RequestBody GradebookDTO gradebook, @PathVariable(
 		
 	}
 	
+	@GetMapping("/assignment/{id}")
+	@Transactional
+	public Assignment getAssignment(@PathVariable int assignment_id) {
+		Assignment assignment = assignmentRepository.findById(assignment_id);
+		
+		return assignment;
+	}
+	
+	//As instructor, add a new assignment for my course.  The assignment has a name and a due date.
+	@PostMapping("/assignment")
+	@Transactional
+
+	public void addAssignment (@RequestParam String name, @RequestParam Date dueDate) {
+		
```

Comment:
What is the reason you used query parameters instead of request body? As I understood request body is more typical for POST requests.

Reply:
Did `@RequestParam Date dueDate` work for you? 
I get this error, maybe my date is not in correct format. 
<img width="661" alt="Screen Shot 2022-09-13 at 12 48 19 PM" src="https://user-images.githubusercontent.com/25463226/189995769-fc3f3e9d-88dc-4849-859a-6a4d15ce08f2.png">

How should the date be sent by a consumer?